### PR TITLE
feat(fig): add spec for 2.0.0

### PR DESCRIPTION
**Pull request by mschrage**
PR generated automatically from push-to-fig-autocomplete-action.

### CODE REVIEW

**Comment by SeparateRecords** [comment]
oh you're kidding, it removed all the generators 😭😭😭😭

**Comment by SeparateRecords** [comment]
am fixing generators now, one-time effort until cli 3.0 😉

Will merge when done

**Comment by SeparateRecords** [comment]
@grant0417 any generators obviously missing?

**Comment by fedeci** [comment]
@SeparateRecords generators removed are a known issue not solvable with the current approach 😕, we need to use ASTs modifications in order to keep them.

File: src/fig/2.0.0.ts [hunk 1]
```diff
@@ -0,0 +1,4146 @@
...
+  membersGenerators,
+  teamsGenerators,
+  workflowsSpecGenerator,
+  sshHostsGenerator,
+  sshIdentityGenerator,
+  userGenerator,
+} from "./shared";
+
+const completion: Fig.Subcommand = {
+  name: "fig_cli",
```
Comment (fedeci) [inline src/fig/2.0.0.ts:17]:
```suggestion
  name: "fig",
```

File: src/fig/2.0.0.ts [hunk 2]
```diff
@@ -0,0 +1,4146 @@
...
+            },
+          ],
+        },
+        {
+          name: "diagnostics",
+          description: "Watches diagnostics",
+          options: [
+            {
+              name: "--rate",
+              isRepeatable: true,
```
Comment (fedeci) [inline src/fig/2.0.0.ts:546]: This does not sound right to me, is this option really repeatable?
Same for a bunch of other options like the ones from the `query-index` subcommand.
```suggestion
              isRepeatable: true,
```

**Review by fedeci — CHANGES_REQUESTED** [review]
@grant0417 can you check the two comments I left, please?

**Comment by fedeci** [comment]
@mschrage I used filemerge to get diffs between the old and new specs. 
Should I build a web tool that calculates diffs from automatic PRs bumping majors and that publishes those diffs to an html accessible by everyone?

**Comment by mschrage** [comment]
@fedeci that would be useful, but I don't think that many people besides us use this tooling - so no need to over optimize it right now!

File: src/fig/2.0.0.ts (hunk 2 above)
Comment (grant0417) [reply src/fig/2.0.0.ts:546]: Yeah, that doesnt look right, hmm, an issue with the clap generator perhaps

Comment (SeparateRecords) [reply src/fig/2.0.0.ts:546]: Let's fix this in an update to the spec - imo for now we should just land this PR. If it's an error in the clap generator, we'll remove it later, if it's in the CLI itself then we'll fix it with semver

**Comment by fedeci** [comment]
> @fedeci that would be useful, but I don't think that many people besides us use this tooling - so no need to over optimize it right now!

Sounds good, eventually I'll implement it in my free time!

File: src/fig/2.0.0.ts (hunk 2 above)
Comment (fedeci) [reply src/fig/2.0.0.ts:546]: Great! @SeparateRecords if everything looks good to you feel free to merge.

**Review by fedeci — APPROVED** [review]

**Comment by grant0417** [comment]
Going to merge the changes!

**Pull request merged**